sw/device/lib/base/status.h

```diff
@@ -56,48 +56,45 @@ typedef struct status {
   int32_t value;
 } status_t;
 
+#define DIF_RESULT_INTO_STATUS(_exp)                                      \
+  ({                                                                      \
+    typeof(_exp) _val = (_exp);                                           \
+    absl_status_t code;                                                   \
+    memcpy(&code, &_val, sizeof(code));                                   \
```

Comment:
nit: Do we have an assertion for the relation between the sizes of `_val` and `code`? If the underlying types are integers can we just add an assertion and cast the result of the expression?

@luismarques @cfrantz

Reply:
This depends on whether or not all arms of `_Generic` have to be valid code considering the possible input types.  When this macro was constructed with `__builtint_types_compatible_p`, all arms of the if statement had to be valid.  You could not cast `struct status` to an integer type like `absl_status_t`, so a `memcpy` is needed to do something equivalent to a C++ `reinterpret_cast<>`.

The assertion _was_ that `__builtin_types_compatible_p` returned true for `dif_result_t`.  Now, the assertion is that this code is only invoked the `_Generic` arm that matches `dif_result_t`.  In both cases, we know that `dif_result_t` is the same size as `absl_status_t`.